pkg/agent/handler/credentialrequestdefinition.go

```diff
@@ -0,0 +1,36 @@
+package handler
+
+import (
+	agentcache "github.com/Axway/agent-sdk/pkg/agent/cache"
+	v1 "github.com/Axway/agent-sdk/pkg/apic/apiserver/models/api/v1"
+	mv1 "github.com/Axway/agent-sdk/pkg/apic/apiserver/models/management/v1alpha1"
+	"github.com/Axway/agent-sdk/pkg/watchmanager/proto"
+)
+
+type crdHandler struct {
+	agentCacheManager agentcache.Manager
+}
+
+// NewCRDHandler creates a Handler for Credential Request Definitions
+func NewCRDHandler(agentCacheManager agentcache.Manager) Handler {
+	return &crdHandler{
+		agentCacheManager: agentCacheManager,
+	}
+}
+
+// Handle processes grpc events triggered for Credentials
+func (h *crdHandler) Handle(action proto.Event_Type, meta *proto.EventMeta, resource *v1.ResourceInstance) error {
```

Comment:
this function is playing tricks on me... 
line 23, will return nil, so why check it?
line 31, if action is created, or updated, why even check for deleted?
line 32, can that return actually be an error?  (didn't check function DeleteCredentialRequestDefinitionByName

Reply:
line 23: This handler is only for credential request definitions, but all handlers get invoked when an event is received.
line 31: you are right, I just mimicked other handlers 
line 32: it can be if there was an issue removing from cache